### ISSUE TITLE
Progress Bar off by default?

### ISSUE THREAD

**Issue by laryn** [backdrop-contrib/webform#139]
Would there be a way to turn the progress bar off by default, and then turn it on automatically if a page break is added to the form? I am always having to go back in and turn it off after saving a form.

**Comment by herbdool** [comment]
Not sure. How is it setup in D7?

**Comment by ghost** [comment]
Yes, I was surprised when I saw it displayed above my form automatically (I've never used it and didn't enable it).

I like the suggestion of enabling it automatically when a 'page break' element is added to the form. But maybe in the Webform settings we can change the 'Show progress bar' checkbox to a select list with the following options:

- Enabled
- Disabled
- Auto

The 'auto' option would be what @laryn suggested above.

**Comment by herbdool** [comment]
There was a recent change in D7 which enabled it by default then a subsequent release where they backtracked. I thought I had updated webform so it would no longer be enabled by default. Can you compare and see what's going on?

**Comment by stpaultim** [comment]
It would appear, according to this issue: https://www.drupal.org/project/webform/issues/2992404, that the default should be set to off for webforms that are only one page. 

It would be great if we could address this. It's really weird to get a progress bar and a simple 3 field contact form.

**Comment by herbdool** [comment]
@stpaultim yup, I've seen that issue.

I'm curious to know what is the default in d7 webform now. Would you have a chance to spin one up and let me know?

**Comment by stpaultim** [comment]
I just did a quick test and I am getting a default progress bar on my single page webform. I guess I am not reading the issue in the issue queue correctly, because I understood that would not be the case. 

I will try to experiment with this a bit more lately, but it appears that single page webforms do have a progress bar by default.

**Comment by stpaultim** [comment]
OK, I did some additional testing and read the issue in the Drupal issue queue closer and I think that they have settled on having the progress bar on by default for all new webforms. Users can opt to change this setting in webform settings. 

I think that "off by default" is a better choice, but this is not an big deal. If we want to stay consistent with Drupal 7, then the status quo is correct. 

I asked for clarification here: https://www.drupal.org/project/webform/issues/2992404#comment-13330199. Just in case I'm misunderstanding, and because I think the issue could use clarification for folks like me.  ;-)

**Comment by olafgrabienski** [comment]
@stpaultim I understand it differently. In the [comment where the D7 webform issue is set to "Fixed"](https://www.drupal.org/project/webform/issues/2992404#comment-12877019), there's the sentence: "People who want progress bars can turn them back on."

**Comment by indigoxela** [comment]
@olafgrabienski It seems, they _think_ it behaves like that, but actually a progress bar appears for a one-page form newly created with the latest webform version on D7.

This might be a Drupal bug (?). In this case - to my opinion - we shouldn't copy the same behavior.

**Comment by olafgrabienski** [comment]
@indigoxela You're right, I've just tested it on simplytest.me with Webform 7.x-4.20, and there is a progress bar on a single page webform. I agree that we shouldn't copy that behavior.

**Comment by laryn** [comment]
Basic PR for testing:
https://github.com/backdrop-contrib/webform/pull/167

**Comment by herbdool** [comment]
I think @olafgrabienski got that incorrectly. In https://www.drupal.org/project/webform/issues/2992404#comment-12877019 they reverted a change which had enabled it for *existing* single page webforms. But it doesn't affect *new* webforms. Here's how it works currently in D7 and in Backdrop:

1. Enable webform
2. Visit global webform settings: `admin/config/content/webform`
3. Progressbar is enabled by default.
4. Create a webform
5. Leave progressbar enabled or disable it on the individual webform.
6. *Or* go back to the global settings and disable progressbar.
7. All new webforms will have progressbar disabled.

So I don't think there's any bug. It was just a change to stop excluding single page webforms if the progressbar setting is enabled. For now I think I prefer keeping it as is, since web admins can just turn it off globally on each site.

**Comment by olafgrabienski** [comment]
> I think @olafgrabienski got that incorrectly.

That's quite possible!

Even if the current behavior isn't a bug, "off by default" seems to be a better choice for some of us. @herbdool do you prefer keeping the progressbar "on by default" for consistency with Drupal 7?

**Comment by herbdool** [comment]
Consistency with D7 is part of it. But also because it is a good default for multi-page webforms. I agree that it's better off by default for single page webforms though I can't see how to make the UX reasonable for those both to be possible.

**Comment by stpaultim** [comment]
> I agree that it's better off by default for single page webforms though I can't see how to make the UX reasonable for those both to be possible.

I understand this problem. It does seem to be a choice about whether it's better to set the default assuming a single or multipage webform. I guess I'm not understanding why it's better to default to an optional setting for multipart webforms?

My viewpoint is that (my assumption here) that the vast majority of webforms are single page forms (I could very well be wrong about this). I very rarely make a multipage webform, so I have to turn this off pretty much every time. I suppose some folks might tend to use multipage webforms much more often. So, for me the more logical default would be to have someone with a multipart webform turn this on, rather than require everyone else to turn it off. 

On most of my sites, I replace the contact form with a webform. Every time I do this, I have to turn off this feature I almost never need. It always surprises me that it is on by default. Often, that is the only webform I create - so by the time I change the global setting on an individual site, I am done making webforms. It also feels very strange to have this on a single page webform, but it doesn't really hurt if it's missing from a multipage webform. It's not really required on a multi-page webform, although it is nice on a multi-page webform.

This makes me think that maybe I'm unusual and that other people create many more multi-page webforms than I do. 

Just trying to clarify why the current default seems strange to me. But, I can live with this either way. I'm starting to get used to it.

**Comment by olafgrabienski** [comment]
> It also feels very strange to have [the progress bar] on a single page webform, but it doesn't really hurt if it's missing from a multipage webform.

That's a good point! (And don't let us get used to the strange default.)

**Comment by olafgrabienski** [comment]
> it is a good default for multi-page webforms. I agree that it's better off by default for single page webforms though I can't see how to make the UX reasonable for those both

I've an idea for a slightly better user interface. Currently, in the Form settings, there is a collapsed fieldset with the *Progress Bar* settings. So, when I'm on the settings page, I don't see at first sight if the Progress Bar is enabled or not.

We could do the following instead:

- Display a single checkbox "Show progress bar" and make it off by default for new installations.
- Show all the other progress bar settings only, when people turn the checkbox on.

**Comment by stpaultim** [comment]
Here is an interesting anecdote from the wild (but only an anecdote). I found this webform on a Drupal 7 site, I am guessing that the site builder is not aware of how easy it is to turn this off. I could be wrong and this was a consious choice. 

But, I find it weird to see this on a single page webform. 

![Contact___ProCreative_Pty_Ltd](https://user-images.githubusercontent.com/3144571/97368006-12791780-1878-11eb-8d21-58a92015d489.png)

**Comment by yorkshire-pudding** [comment]
I would agree that off by default seems the better option.  As the setting is stored per webform, there is no danger of changing existing webforms, whether for existing Backdrop users or people upgrading from Drupal 7.  The PR looks good and should carry over the setting if it is set in D7, while not enabling it for new installs.
